### ISSUE TITLE
Parse error messages are displayed twice.

### ISSUE THREAD

**Issue by deathaxe** [rosshadden/sublime-xpath#32]
XPath adds the error messages from lxml twice to the statusbar. This is a bit confusing and a waste of valuable space.

![screenshot](https://cloud.githubusercontent.com/assets/16542113/26077659/1f0c3c2c-39bd-11e7-9f29-c12c44037d6e.png)

_Note: Exalt was disabled after realizing the message 3 times on the statusbar ;-)_

Could you please shorten output or even add an option to disable it?

Disabling error messages at all would help to improve 	interoperability with Exalt, which is used for validation. It outputs the same messages on parse/validation errors.  Or can XPath add_region(...) to highlight parse/validation errors to need only one package to handle XML?

**Comment by keith-hall** [comment]
I'll have a look at implementing the following:
- only display errors once
- option to hide errors (or maybe automatically hide them when Exalt is installed?)
- option to highlight error region

I'm not sure how to go about shortening errors other than not duplicating them... ;) Maybe an option to remove error position from the status bar?

**Comment by deathaxe** [comment]
Removing duplicate error messages is very fine. XPath should either show path or error message or nothing. I would prefer an option to disable message output to prevent inter-package dependencies.

Same with highlighted regions. Highlighting parse or validation errors makes sense only, if no other package (SublimeLinter or Exalt) handles these errors. I am not sure whether validation is already part of XPath?

Are _line_ and _column_ provided by lxml itself? In my opinion it is a bit confusing to have  these information just next to normal text position. It is not clearly visible they belong to the error.

**Comment by keith-hall** [comment]
@deathaxe regarding the appearance of and ability to hide the parse errors, I've submitted a PR at https://github.com/rosshadden/sublime-xpath/pull/33 which should also answer most of your questions. To clarify, this plugin doesn't validate an XML document in terms of DTD schemas etc., but if the document is not well formed, `lxml` will throw a parse error, which this plugin will (now optionally not) display in the status bar.

I'll work on adding an option to highlight parse errors as part of a separate PR.

**Comment by deathaxe** [comment]
I checked #33 out and disabled error messages. With _Exalt_ enabled, too, everything is fine.

As parse errors are highlighted by _Exalt_ already and scheme validation is not part of _XPath_ I think it is not necessary to add any highlighting feature to XPath. It would make sense only, if _Exalt_ was not required anymore.

The idea behind was to parse the XML structure only once by one plugin performing both on a common dataset - validation and path handling.

**Comment by keith-hall** [comment]
I agree that it doesn't make sense to add error highlighting to this package when Exalt was designed for it and does a more thorough job, in which case I believe we can consider this issue resolved once the PR is merged.

Yeah, it's a shame that the XML structure gets parsed separately by each plugin - but there's no current way round that without merging the packages. It'll be nice if someone could build an LSP server implementation for XML that would work with XPath and Schema validation, and a Sublime LSP client that would re-use the same LSP server from both plugins (which would only need to keep state once as it is the same document buffer).

I briefly mentioned other advantages such an implementation would have at https://github.com/rosshadden/sublime-xpath/issues/9#issuecomment-289256401, mainly allowing us to no longer need a million workarounds for `lxml`s poor tag location support (massively simplifying this package and reducing the millions of potential failure points) and potentially be able to issue XPath 2.0 queries ;)

**Comment by deathaxe** [comment]
On the other hand an LSP server needs to sync with the content to keep track of modifications. I read about partial updates but I am still skeptic about it. If all plugins will use such servers in the future, we'll have 10 servers per document, which all need to be feet with the text content and thus will keep separate structures too - but vice versa. Not sure what's worse?

Basically I like the idea of "native" python packages being used to handle things locally. But maybe I am only too old fashioned to become a friend of all these heavy server client architectures being used everywhere.

I didn't dig into lxml too much, so can't say anything about its pitfalls. I only read about the claims to be much more user/dev friendly than the native libxml2 library.
